vispy/visuals/colorbar.py

```diff
@@ -0,0 +1,223 @@
+    orientation : {'horizontal', 'vertical'}
+        the orientation of the colorbar, used for coloring
+
+            * 'horizontal': the color is applied from left to right,
+              with minimum corresponding to left and maximum to right
+            * 'vertical': the color is applied from bottom to top,
+              with mimumum corresponding to bottom and maximum to top
+    """
+
+    def __init__(self, center_pos, halfdim,
+                 label,
+                 cmap,
+                 clim=(0.0, 1.0),
+                 orientation="horizontal",
+                 **kwargs):
+
+        super(ColorBarVisual, self).__init__(**kwargs)
+
+        self._cmap = get_colormap(cmap)
+        self._clim = clim
+
+        x, y = center_pos
+        halfw, halfh = halfdim
+
+        vertices = np.array([[x - halfw, y - halfh],
+                            [x + halfw, y - halfh],
+                            [x + halfw, y + halfh],
+                             # tri 2
+                            [x - halfw, y - halfh],
+                            [x + halfw, y + halfh],
+                            [x - halfw, y + halfh]],
+                            dtype=np.float32)
+
+        tex_coords = np.array([[0, 0], [1, 0], [1, 1],
+                              [0, 0], [1, 1], [0, 1]],
+                              dtype=np.float32)
+
+        if orientation == "horizontal":
+            self._program = ModularProgram(VERT_SHADER, FRAG_SHADER_HORIZONTAL)
+
+        elif orientation == "vertical":
+            self._program = ModularProgram(VERT_SHADER, FRAG_SHADER_VERTICAL)
+
+        else:
+            raise ValueError("orientation must"
+                             " be \'horizontal\' or \'vertical\'."
+                             " given: %s" % orientation)
+
+        self._program.frag['color_transform'] = Function(self._cmap.glsl_map)
+        self._program['a_position'] = vertices.astype(np.float32)
+        self._program['a_texcoord'] = tex_coords.astype(np.float32)
+
+        if orientation == "horizontal":
+            text_x, text_y = x, y + halfh * 1.2
+            self._label = TextVisual(label, pos=(text_x, text_y),
+                                     anchor_y="top")
+
+            begin_tick_pos = x - halfw, text_y
+            end_tick_pos = x + halfw, text_y
+
+            # TODO, HACK: This should ideally be a single TextVisual
+            # However, one TextVisual with multiple strings
+            # does not seem to be working as of now. (See #981)
+            # https://github.com/vispy/vispy/issues/981
+            self._ticks = []
+            self._ticks.append(TextVisual(str(clim[0]),
+                               pos=begin_tick_pos,
+                               anchor_y="top"))
+            self._ticks.append(TextVisual(str(clim[1]),
+                               pos=end_tick_pos,
+                               anchor_y="top"))
+
+        elif orientation == "vertical":
```

Comment:
I recognize that there is code duplication here, but I prefer the explicitness of it, since the two halves of the code setup the colorbar differently. (```rotation``` is setup, along with different positions for the markers).

I'd like your opinions.